src/main/java/org/springframework/integration/kafka/dsl/KafkaInboundChannelAdapterSpec.java

```diff
@@ -31,33 +32,51 @@
  * @param <V> the value type.
  *
  * @author Gary Russell
+ * @author Anshul Mehra
  *
  * @since 3.0.1
  *
  */
 public class KafkaInboundChannelAdapterSpec<K, V>
 		extends MessageSourceSpec<KafkaInboundChannelAdapterSpec<K, V>, KafkaMessageSource<K, V>> {
 
+	@Deprecated
 	KafkaInboundChannelAdapterSpec(ConsumerFactory<K, V> consumerFactory, boolean allowMultiFetch, String... topics) {
-		this.target = new KafkaMessageSource<>(consumerFactory, allowMultiFetch, topics);
+		this.target = new KafkaMessageSource<>(consumerFactory, new ConsumerProperties(topics), allowMultiFetch);
 	}
 
+	@Deprecated
 	KafkaInboundChannelAdapterSpec(ConsumerFactory<K, V> consumerFactory,
 			KafkaAckCallbackFactory<K, V> ackCallbackFactory, boolean allowMultiFetch, String... topics) {
 
-		this.target = new KafkaMessageSource<>(consumerFactory, ackCallbackFactory, allowMultiFetch, topics);
+		this.target = new KafkaMessageSource<>(consumerFactory, new ConsumerProperties(topics), ackCallbackFactory, allowMultiFetch);
 	}
 
+	KafkaInboundChannelAdapterSpec(ConsumerFactory<K, V> consumerFactory,
+			ConsumerProperties consumerProperties, boolean allowMultiFetch) {
+		this.target = new KafkaMessageSource<>(consumerFactory, consumerProperties, allowMultiFetch);
+	}
+
+	KafkaInboundChannelAdapterSpec(ConsumerFactory<K, V> consumerFactory,
+			ConsumerProperties consumerProperties,
+			KafkaAckCallbackFactory<K, V> ackCallbackFactory, boolean allowMultiFetch) {
+
+		this.target = new KafkaMessageSource<>(consumerFactory, consumerProperties, ackCallbackFactory, allowMultiFetch);
+	}
+
+	@Deprecated
```

Comment:
Please, also add JavaDocs into all the `public` API with the `@deprecated` tag and explanation in favor of what it is deprecated.

Thanks